### PR TITLE
BBUS sleeping bag respawn point

### PR DESCRIPTION
**Pull request by Basipek**
I wrote the sleeping bag entity, works as a spawn point like in JMod-Jegacy.
Turned it into a single file for uniformity and added an Alt+Use section, no functionality to that yet.

What else shall we do? Is there anything to fix before?
GMT+3 btw, way past sleepy time at the time of writing.

### CODE REVIEW

File: lua/entities/ent_jack_sleepingbag.lua [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			phys:SetMass(35)
+			self:SetColor(Color(100,100,100))
+		end
+		
+		self:SetUseType(SIMPLE_USE)
+		
+		self.nextSpawnTime=0
+	end
+
+	function ENT:Use(a)
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:33]: the standard we use in jmod is `ply`, like garry. So just change `a` to `ply`

File: lua/entities/ent_jack_sleepingbag.lua [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				if (a~=self.jackSleepingBagOwner) then
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already claimed!")
+				else
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already yours!")
+				end
+			else
+				if (IsValid(a.jackSleepingBag)) then a.jackSleepingBag.jackSleepingBagOwner=nil;a.jackSleepingBag:SetColor(Color(100,100,100)) end
+				a:PrintMessage(HUD_PRINTCENTER,"Bed has been claimed!")
+				self.jackSleepingBagOwner=a
+				a.jackSleepingBag=self
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:48]: change `.jackSleepingBag` to `.JModSpawnPointEntity` to properly distinguish it as part of of this addon

File: lua/entities/ent_jack_sleepingbag.lua [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		elseif a:IsPlayer() then 
+			if (self.jackSleepingBagOwner and IsValid(self.jackSleepingBagOwner)) then
+				if (a~=self.jackSleepingBagOwner) then
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already claimed!")
+				else
+					a:PrintMessage(HUD_PRINTCENTER,"This bed is already yours!")
+				end
+			else
+				if (IsValid(a.jackSleepingBag)) then a.jackSleepingBag.jackSleepingBagOwner=nil;a.jackSleepingBag:SetColor(Color(100,100,100)) end
+				a:PrintMessage(HUD_PRINTCENTER,"Bed has been claimed!")
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:46]: "Bed has been claimed" could be interpreted as "someone else already claimed this bed"
so change it to "You've claimed this bed!" or something like that

File: lua/entities/ent_jack_sleepingbag.lua [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+
+if (CLIENT) then
+	function ENT:Draw()
+		self.Entity:DrawModel()
+	end
+elseif (SERVER) then
+
+	function ENT:Initialize()
+		self.Entity:SetModel("models/props_equipment/sleeping_bag1.mdl")
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:16]: if this comes from Left 4 Dead, then we need to port this model into JMod. We can't rely on people having JMod-Legacy installed just for this to work. JMod-Legacy is basically just an optional DLC for JMod. Gotta get both the model and the material in here

File: lua/entities/ent_jack_sleepingbag.lua [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		self.Entity:PhysicsInit(SOLID_VPHYSICS)
+		self.Entity:SetMoveType(MOVETYPE_VPHYSICS)
+		self:SetSolid( SOLID_VPHYSICS )
+		self.jackSleepingBagOwner=nil
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:20]: 1) in all cases where we assign the owner, instead use `JMod.SetEZowner(ply)`
2) in all cases where we check the owner, instead check for `.EZowner`

this goes for the rest of the PR. `EZowner` is the standard we use in JMod

**Comment by Jackarunda** [comment]
Consider turning all the `PrintMessage`s into `JMod.Hint(` calls, like the rest of the addon uses, so that they can be properly translated later.
Also, add a crafting recipe and a radio drop order for the sleeping bag so people can get them without the spawnmenu.
Overall, good work, just needs some more things. I'll help in #gmod-development as needed

File: lua/entities/ent_jack_sleepingbag.lua [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+		if((data.Speed>80)and(data.DeltaTime>0.2))then
+			self.Entity:EmitSound("Body.ImpactSoft")
+		end
+	end
+
+	function ENT:OnRemove()
+		if(IsValid(self.EZowner))then self.EZowner.JModSpawnPointEntity=nil end
+	end
+
+	local function jackSpawnHook(ply)
```
Comment (Jackarunda) [inline lua/entities/ent_jack_sleepingbag.lua:73]: take the contents of this function and move them into `JackaSpawnHook`, in `sv_init.lua`, for centralization

File: lua/jmod/sv_init.lua [hunk 7]
```diff
@@ -51,6 +51,18 @@ local function JackaSpawnHook(ply)
 		end
 	end
 
+	if((ply.JModSpawnPointEntity)and(IsValid(ply.JModSpawnPointEntity)))then
+		if(ply.JModSpawnPointEntity.nextSpawnTime<CurTime())then
+			ply.JModSpawnPointEntity.nextSpawnTime=CurTime()+60
+			ply:SetPos(ply.JModSpawnPointEntity:GetPos())
```
Comment (AdventureBootCoder) [inline lua/jmod/sv_init.lua:57]: I feel like this might get the player stuck in something

**Pull request merged**